src/Pages/Applications/DevtronApps/Details/AppConfigurations/MainContent/DeploymentTemplate/DeploymentTemplate.tsx

```diff
@@ -0,0 +1,1640 @@
+        showDeleteOverrideDialog,
+        isSaving,
+        showSaveChangesModal,
+        isLoadingChangedChartDetails,
+        isLoadingInitialData,
+        initialLoadError,
+        resolvedPublishedTemplate,
+    } = state
+
+    const manifestAbortController = useRef<AbortController>(new AbortController())
+    const [, grafanaModuleStatus] = useAsync(() => getModuleInfo(ModuleNameMap.GRAFANA), [])
+
+    const isDryRunView = configHeaderTab === ConfigHeaderTabType.DRY_RUN
+    const isInheritedView = configHeaderTab === ConfigHeaderTabType.INHERITED
+    const isDraftAvailable = isProtected && !!draftTemplateData?.latestDraft
+
+    const isPublishedValuesView = !!(
+        isDraftAvailable &&
+        ((configHeaderTab === ConfigHeaderTabType.VALUES &&
+            selectedProtectionViewTab === ProtectConfigTabsType.PUBLISHED) ||
+            (configHeaderTab === ConfigHeaderTabType.DRY_RUN && dryRunEditorMode === DryRunEditorMode.PUBLISHED_VALUES))
+    )
+    const isCompareView = !!(
+        isProtected &&
+        configHeaderTab === ConfigHeaderTabType.VALUES &&
+        selectedProtectionViewTab === ProtectConfigTabsType.COMPARE
+    )
+
+    const isApprovalPending = isDraftAvailable && draftTemplateData.latestDraft.draftState === DraftState.AwaitApproval
+    const isApprovalView =
+        isApprovalPending && (isCompareView || (isDryRunView && dryRunEditorMode === DryRunEditorMode.APPROVAL_PENDING))
+
+    const showApprovalPendingEditorInCompareView =
+        isCompareView &&
+        isApprovalView &&
+        compareFromSelectedOptionValue === CompareFromApprovalOptionsValuesType.APPROVAL_PENDING
+
+    const showNoOverrideTab =
+        !!envId && !isDraftAvailable && !publishedTemplateData?.isOverridden && !currentEditorTemplateData?.isOverridden
+
+    const showNoOverrideEmptyState = showNoOverrideTab && configHeaderTab === ConfigHeaderTabType.VALUES
+    // TODO: After CM/CS Merge strategy is implemented, we can re-use enum
+    const isDeleteOverrideDraft = !!envId && draftTemplateData?.latestDraft?.action === 3
+    const showDeleteOverrideDraftEmptyState =
+        isDeleteOverrideDraft &&
+        configHeaderTab === ConfigHeaderTabType.VALUES &&
+        selectedProtectionViewTab === ProtectConfigTabsType.EDIT_DRAFT
+
+    const isLoadingSideEffects = isResolvingVariables || isSaving || isLoadingChangedChartDetails
+
+    /**
+     * There are two cases:
+     * 1. In case of base config - Published config is the one that we are using currently and is always present (Even in zero state we are have a default saved config)
+     * 2. In case of override - Published config is the one that is overridden from base config (Inherited) and is not always present
+     */
+    const isPublishedConfigPresent = !(envId && !publishedTemplateData?.isOverridden)
+
+    const showNoPublishedVersionEmptyState = isPublishedValuesView && !isPublishedConfigPresent
+
+    const isEditMode = isProtected
```

Comment:
We can write this as
```suggestion
    const isEditMode = configHeaderTab === ConfigHeaderTabType.VALUES && (!isProtected || selectedProtectionViewTab === ProtectConfigTabsType.EDIT_DRAFT)
```